src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Ball {
+    private String color;
+    private int number;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
Having an instance of ColorSupplier as a class field in the Ball class is not necessary and can lead to tight coupling. Consider passing the color as a parameter to the setColor method or using the ColorSupplier where you actually need to generate a random color.